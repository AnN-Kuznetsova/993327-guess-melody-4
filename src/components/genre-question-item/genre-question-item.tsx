import * as React from "react";

import {AnswerGenreType} from "../../types";


interface Props {
  answer: AnswerGenreType,
  id: number;
  onChange: (id: number, value: boolean) => void;
  renderPlayer: (src: string, id: number) => React.ReactNode;
  userAnswer: boolean;
};


export const GenreQuestionItem: React.FunctionComponent<Props> = (props: Props) => {
  const {
    answer,
    id,
    onChange,
    renderPlayer,
    userAnswer,
  } = props;

  const handleUserAnswerChange = (event) => {
    const value = event.target.checked;
    const index = event.target.dataset.answerIndex;
    onChange(index, value);
  };

  return (
    <div className="track" key={`${id}-${answer.src}`}>
      {renderPlayer(answer.src, id)}

      <div className="game__answer">
        <input
          className="game__input visually-hidden"
          type="checkbox"
          name="answer"
          value={`answer-${id}`}
          id={`answer-${id}`}
          data-answer-index={id}
          checked={userAnswer}
          onChange={handleUserAnswerChange}
        />
        <label className="game__check" htmlFor={`answer-${id}`}>Отметить</label>
      </div>
    </div>
  );
};

