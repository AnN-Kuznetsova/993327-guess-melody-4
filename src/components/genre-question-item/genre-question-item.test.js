import React from "react";
import renderer from "react-test-renderer";

import {GenreQuestionItem} from "./genre-question-item";


const props = {
  answer: {
    src: `answer-src`,
    genre: `answer-genre`,
  },
  id: 0,
  onChange: () => {},
  renderPlayer: () => {},
  userAnswer: false,
};


describe(`Render GenreQuestionItem`, () => {
  it(`Should match with snapshot`, () => {
    const genreQuestionItemSnapshot = renderer.create(
        <GenreQuestionItem {...props} />
    ).toJSON();

    expect(genreQuestionItemSnapshot).toMatchSnapshot();
  });
});
