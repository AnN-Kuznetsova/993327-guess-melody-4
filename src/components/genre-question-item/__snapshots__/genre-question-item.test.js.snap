// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render GenreQuestionItem Should match with snapshot 1`] = `
<div
  className="track"
>
  <div
    className="game__answer"
  >
    <input
      checked={false}
      className="game__input visually-hidden"
      data-answer-index={0}
      id="answer-0"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-0"
    />
    <label
      className="game__check"
      htmlFor="answer-0"
    >
      Отметить
    </label>
  </div>
</div>
`;
